tests/fixtures/script_deployment_fixture.py

```diff
@@ -1,68 +1,64 @@
+the command line in the context of tests). Unfortunately, the names and even the meanings
+of some of the parameters in those two scenarios do not match.
 
+At some point, we will standardise the names and replace the deploy_params and upload_params
+fixtures with backend_aware_database_params and backend_aware_bucketfs_params.
+"""
+from __future__ import annotations
 from typing import Any
 from urllib.parse import urlparse
-
 import pytest
-from exasol.pytest_itde import config
-
-from tests.fixtures.database_connection_fixture_constants import BACKEND_ONPREM, BACKEND_SAAS
-from tests.utils.parameters import bucketfs_params
+from exasol.pytest_backend import BACKEND_ONPREM
 
 
-@pytest.fixture(scope="session")
-def deploy_params_onprem(exasol_config: config.Exasol) -> dict[str, Any]:
-    return {
-        'dsn': f"{exasol_config.host}:{exasol_config.port}",
-        'db_user': exasol_config.username,
-        'db_pass': exasol_config.password
-    }
+_deploy_param_map = {
+    'dsn': 'dsn',
+    'user': 'db_user',
+    'password': 'db_pass'
+}
 
+_upload_param_map = {
+    'username': 'bucketfs-user',
+    'password': 'bucketfs-password',
+    'service_name': 'bucketfs-name',
+    'bucket_name': 'bucket',
+    'url': 'saas_url',
+    'account_id': 'saas_account_id',
+    'database_id': 'saas_database_id',
+    'pat': 'saas_token'
+}
 
-@pytest.fixture(scope="session")
-def upload_params_onprem(bucketfs_config: config.BucketFs):
-    parsed_url = urlparse(bucketfs_config.url)
+def _parse_bucketfs_url(url: str) -> dict[str, Any]:
+    parsed_url = urlparse(url)
     host, port = parsed_url.netloc.split(":")
     return {
-        "bucketfs-name": bucketfs_params.name,
         "bucketfs-host": host,
         "bucketfs-port": port,
-        "bucketfs-use-https": False,
-        "bucketfs-user": bucketfs_config.username,
-        "bucketfs-password": bucketfs_config.password,
-        "bucket": bucketfs_params.bucket
+        "bucketfs-use-https": parsed_url.scheme.lower() == 'https',
     }
 
 
-@pytest.fixture(scope="session")
-def deploy_params_saas(saas_url, saas_account_id, saas_database_id, saas_token) -> dict[str, Any]:
-    yield {
-        'saas_url': saas_url,
-        'saas_account_id': saas_account_id,
-        'saas_database_id': saas_database_id,
-        'saas_token': saas_token
-    }
+def _translate_params(source: dict[str, Any], param_map: dict[str, str]) -> dict[str, Any]:
+    return {param_map[k]: v for k, v in source.items() if k in param_map}
 
 
 @pytest.fixture(scope="session")
-def deploy_params(backend,
-                  deploy_params_onprem,
-                  deploy_params_saas) -> dict[str, Any]:
-    if backend == BACKEND_ONPREM:
-        yield deploy_params_onprem
-    elif backend == BACKEND_SAAS:
-        yield deploy_params_saas
-    else:
-        raise ValueError(f'No deploy_params fixture for the backend {backend}')
+def deploy_params(backend_aware_database_params) -> dict[str, Any]:
```

Comment:
Wouldn't it be easier for the clients if the fixture returns a dataclass or namedtuple, instead of a dictionay?

Reply:
For the clients, it would be easier if these fixtures didn't exist :). I think the dictionary is fine, it can be used as a **kwargs. Also, depending on the solution we come up with, we may replace these fixtures with `backend_aware_database_params` and `backend_aware_bucketfs_params`, which are dictionaries.